driver-core/src/main/com/mongodb/connection/ClusterSettings.java

```diff
@@ -550,7 +553,18 @@ private ClusterSettings(final Builder builder) {
 
         srvHost = builder.srvHost;
         hosts = builder.hosts;
-        mode = builder.mode != null ? builder.mode : hosts.size() == 1 ? ClusterConnectionMode.SINGLE : ClusterConnectionMode.MULTIPLE;
+        if (srvHost != null) {
+            if (builder.mode == ClusterConnectionMode.SINGLE) {
+                throw new IllegalArgumentException("An SRV host name was provided but the connection mode is not MULTIPLE");
+            }
+            mode = ClusterConnectionMode.MULTIPLE;
+        } else {
+            if (builder.mode == ClusterConnectionMode.SINGLE && builder.hosts.size() > 1) {
```

Comment:
Just moved this code from line 538 to consolidate it near the setting of the mode